lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -249,6 +249,39 @@ def test_rebinding_same_user(self, block_type):
         render.get_module_for_descriptor(self.mock_user, request, descriptor, field_data_cache, self.toy_course.id)
         render.get_module_for_descriptor(self.mock_user, request, descriptor, field_data_cache, self.toy_course.id)
 
+    def test_rebind_different_users(self):
+        request = self.request_factory.get('')
+        request.user = self.mock_user
+        course = CourseFactory()
+        descriptor = ItemFactory(category='html', parent=course)
+        # at this point, descriptor shouldn't have _unwrapped_field_data defined on it
+        with self.assertRaises(AttributeError):
+            descriptor._unwrapped_field_data
+
+        field_data_cache = FieldDataCache(
+            [self.toy_course, descriptor], self.toy_course.id, self.mock_user
+        )
+        render.get_module_for_descriptor(
+            self.mock_user, request, descriptor, field_data_cache, self.toy_course.id
+        )
+        # at the point, descriptor._unwrapped_field_data should be set
+        descriptor._unwrapped_field_data
```

Comment:
Check that this is the same `_field_data` that it had before you called `get_module_for_descriptor`